### ISSUE TITLE
Make upload and download work offline

### ISSUE THREAD

**Issue by bollwyvl** [deathbeds/jyve#37]
Drag-to-upload (or the upload button) and download should work.

This probably means _actually_ solving the contents problem.

**Comment by SimonBiggs** [comment]
What about dexie js as a wrapper around IndexedDB. Creating a contents manager that uses the inbrowser web interface?

**Comment by bollwyvl** [comment]
yeah, custom contents manager/fs is the right approach. pretty easy to drop
in an IDrive implementation, though the ui might be a bit unexpected.

a long-term goal is something that (also) works with IPFS and dat, but
indexedb, serviceworker, localstorage is probably a fine initial approach.

either way, getting the kernels access to this stuff is also important, but
just being able to get _a changed notebook_ in and out is the biggest value
in the near term.

**Comment by SimonBiggs** [comment]
If Dexiejs was used as an IDrive implementation then the following tool might be used to sync to a server if desired while still keeping a principle of supporting the static use case as front and centre.

http://dexie.org/docs/Syncable/Dexie.Syncable.js.html
